code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -31,13 +31,21 @@ var/list/ai_list = list()
 	var/viewalerts = 0
 	var/icon/holo_icon//Default is assigned when AI is created.
 	var/obj/mecha/controlled_mech //For controlled_mech a mech, to determine whether to relaymove or use the AI eye.
-	var/radio_enabled = 1 //Determins if a carded AI can speak with its built in radio or not.
+	var/radio_enabled = 1 //Determines if a carded AI can speak with its built in radio or not.
 	radiomod = ";" //AIs will, by default, state their laws on the internal radio.
 	var/obj/item/device/pda/ai/aiPDA = null
 	var/obj/item/device/multitool/aiMulti = null
 	var/obj/machinery/bot/Bot
 	var/tracking = 0 //this is 1 if the AI is currently tracking somebody, but the track has not yet been completed.
 
+
+	//RESEARCH
+	var/researchrate = 1
+	var/researchtime
+	var/researching = 0
+	var/list/subsystems = list("doors"=1,"lights"=1,"comms"=1) //shows subsystems AI can use

```

Comment:
"It's done elsewhere in the code" != "It's acceptable"

You should be using ```#defines```, yes they'll be strings in the back end, but it prevents typos.